logger/common.go

```diff
@@ -334,6 +371,18 @@ func (l *Logger) Read(
 	}
 }
 
+// generateRandomID is based on Docker
+// GenerateRandomID: https://github.com/moby/moby/blob/bca8d9f2ce0d63e1490692917cde6273bc288bad/pkg/stringid/stringid.go#L40
```

Comment:
Why can we not just use that method? Why are we replicating it here?

Reply:
This is explained in my comment below this line... I can make it longer and more detailed to make it clearer. The Docker generate function does the rand generation in a loop, because it can't allow the ID to container only 0 - 9 (digits) because then it can't be used a container ID. We don't have that concern.

Reply:
Another reason to not import from docker is properly handling that error return in `rand.Read`